src/server/session.ts

```diff
@@ -601,20 +601,58 @@ namespace ts.server {
             }
 
             if (simplifiedResult) {
-                return definitions.map(def => {
-                    const defScriptInfo = project.getScriptInfo(def.fileName);
-                    return {
-                        file: def.fileName,
-                        start: defScriptInfo.positionToLineOffset(def.textSpan.start),
-                        end: defScriptInfo.positionToLineOffset(textSpanEnd(def.textSpan))
-                    };
-                });
+                return this.getSimplifiedDefinitions(definitions, project);
             }
             else {
                 return definitions;
             }
         }
 
+        private getDefinitionAndBoundSpan(args: protocol.FileLocationRequestArgs, simplifiedResult: boolean): protocol.DefinitionInfoAndBoundSpan | DefinitionInfoAndBoundSpan {
+            const { file, project } = this.getFileAndProject(args);
+            const position = this.getPositionInFile(args, file);
+            const scriptInfo = project.getScriptInfo(file);
+
+            const definitionAndBoundSpan = project.getLanguageService().getDefinitionAndBoundSpan(file, position);
+
+            if (!definitionAndBoundSpan || !definitionAndBoundSpan.definitions) {
+                return {
+                    definitions: emptyArray,
+                    textSpan: undefined
+                };
+            }
+
+            if (simplifiedResult) {
+                return {
+                    definitions: this.getSimplifiedDefinitions(definitionAndBoundSpan.definitions, project),
+                    textSpan: this.getSimplifiedTextSpan(scriptInfo, definitionAndBoundSpan.textSpan)
+                };
+            }
+
+            return definitionAndBoundSpan;
+        }
+
+        private getSimplifiedDefinitions(definitions: ReadonlyArray<DefinitionInfo>, project: Project): ReadonlyArray<protocol.FileSpan> {
+            return definitions.map(def => this.getSimplifiedFileSpan(def.fileName, def.textSpan, project));
+        }
+
+        private getSimplifiedFileSpan(fileName: string, textSpan: TextSpan, project: Project): protocol.FileSpan {
+            const scriptInfo = project.getScriptInfo(fileName);
+            const simplifiedTextSpan = this.getSimplifiedTextSpan(scriptInfo, textSpan);
+
+            return {
+                file: fileName,
+                ...simplifiedTextSpan
+            };
+        }
+
+        private getSimplifiedTextSpan(scriptInfo: ScriptInfo, textSpan: TextSpan): protocol.TextSpan {

```

Comment:
>getSimplifiedTextSpan [](start = 16, length = 21)

Should this be called in more places?

Reply:
Maybe, there must be some other places where a "simplified textSpan" might be required and this method could be reused. I didn't check for that thou. 

In any case, this was refactored to reuse in two different places. line 628 and line 641.

Reply:
Refactored for reusability. Also based on a conversation with @mhegazy, changed a couple of method names.